zilliqa/src/p2p_node.rs

```diff
@@ -260,8 +259,8 @@ impl P2pNode {
 
                     let topic = Self::shard_id_to_topic(shard_id);
 
-                    // Push messages back into queue if there are no peers
-                    if self.swarm.behaviour().gossipsub.all_peers().collect_vec().is_empty() {
+                    // Push messages back into queue if there are no peers and it is not to self
+                    if self.swarm.behaviour().gossipsub.all_peers().collect_vec().is_empty() && dest != Some(from) && dest.is_some()  {
```

Comment:
nit: The `.is_some()` is unnecessary and I think its easier to read the logic if we exclude self-sends first.
```suggestion
                    if dest != Some(from) && self.swarm.behaviour().gossipsub.all_peers().collect_vec().is_empty() {
```